### ISSUE TITLE
RF-DETR postprocessing fails when num_select >= available candidates

### ISSUE THREAD

**Issue by MNakahara-ishidatec** [CVHub520/X-AnyLabeling#1449]
### Description

RF-DETR inference can fail when `num_select` is greater than or equal to the number of available candidates in the model output.

In [anylabeling/services/auto_labeling/rfdetr.py](anylabeling/services/auto_labeling/rfdetr.py), the postprocess logic does:

```python
topk_indexes = np.argpartition(
    -prob_reshaped, self.num_select, axis=1
)[:, : self.num_select]
```

This is problematic because `np.argpartition` requires `kth` to be strictly smaller than the number of flattened logits. If `num_select` is equal to or greater than the candidate count, NumPy raises an out-of-bounds error.

### Steps to reproduce

1. Configure RF-DETR with a `num_select` value that is equal to or greater than the number of available candidates
2. Run inference with a model output whose flattened logits count is smaller than or equal to that value
3. The model crashes during postprocessing

### Expected behavior

Inference should continue normally and use only the available candidates instead of failing during top-k selection.

### Actual behavior

The process raises a NumPy `ValueError` during `argpartition`, causing inference to fail.

**Comment by MNakahara-ishidatec** [comment]
Sorry, I misidentified the source of the issue.

After further investigation, this issue is caused by the implementation in the upstream `anylabeling` module, specifically `anylabeling/services/auto_labeling/rfdetr.py`, rather than by the changes in this fork.

I will close this issue and report it to the upstream repository instead.

Sorry for the confusion.

**Comment by MNakahara-ishidatec** [comment]
Sorry for the confusion.

I mistakenly concluded that this issue originated in the upstream `anylabeling` implementation.

After checking again, I confirmed that the issue is specific to X-AnyLabeling, so this issue is valid here.

I'm reopening it.

**Comment by CVHub520** [comment]
Confirmed and fixed in main via #1450. Verified with a real RF-DETR Nano ONNX model. Thanks for reporting!
